source/plugins/go/input/perf/perf.go

```diff
@@ -0,0 +1,203 @@
+	hostName                  = ""
+	isWindows                 = false
+	telemetryTimeTracker      int64
+	isFromCache	       = false
+)
+
+// Init An instance of the configuration loader will be passed to the Init method so all the required
+// configuration entries can be retrieved within the plugin context.
+func (p *perfPlugin) Init(ctx context.Context, fbit *plugin.Fluentbit) error {
+	if fbit.Conf.String("tag") == "" {
+		p.tag = tag
+	} else {
+		p.tag = fbit.Conf.String("tag")
+	}
+	if fbit.Conf.String("run_interval") == "" {
+		p.runInterval = runInterval
+	} else {
+		p.runInterval, _ = strconv.Atoi(fbit.Conf.String("run_interval"))
+	}
+	if fbit.Conf.String("insightsmetricstag") == "" {
+		p.insightsmetricstag = insightsmetricstag
+	} else {
+		p.insightsmetricstag = fbit.Conf.String("insightsmetricstag")
+	}
+
+	osType := os.Getenv("OS_TYPE")
+	if strings.EqualFold(osType, "windows") {
+		FLBLogger = lib.CreateLogger("/etc/amalogswindows/fluent-bit-input.log")
+	} else {
+		FLBLogger = lib.CreateLogger("/var/opt/microsoft/docker-cimprov/log/fluent-bit-input.log")
+	}
+	return nil
+}
+
+// Collect this method will be invoked by the fluent-bit engine after the initialisation is successful
+// this method can lock as each plugin its implemented in its own thread. Be aware that the main context
+// can be cancelled at any given time, so handle the context properly within this method.
+// The *ch* channel parameter, is a channel handled by the runtime that will receive messages from the plugin
+// collection, make sure to validate channel closure and to follow the `plugin.Message` struct for messages
+// generated by plugins.
+func (p perfPlugin) Collect(ctx context.Context, ch chan<- plugin.Message) error {
+	tick := time.NewTicker(time.Duration(p.runInterval) * time.Second)
+
+	for {
+		select {
+		case <-ctx.Done():
+			err := ctx.Err()
+			if err != nil && !errors.Is(err, context.Canceled) {
+				return err
+			}
+
+			return nil
+		case <-tick.C:
+			emitTime := time.Now()
+			telemetryTimeTracker = emitTime.Unix()
+			FLBLogger.Print("perf::enumerate.start @ ", time.Now().UTC().Format(time.RFC3339))
+			perfmessages, insightsmetricsmessages := p.enumerate()
+			FLBLogger.Print("perf::enumerate.end @ ", time.Now().UTC().Format(time.RFC3339))
+
+			ch <- plugin.Message{
+				Record: map[string]any {
+					"tag":     tag,
+					"messages": perfmessages,
+				},
+				Time: emitTime,
+			}
+			FLBLogger.Print("perf::emitted ", len(perfmessages) ," perf records @ ", time.Now().UTC().Format(time.RFC3339))
+			ch <- plugin.Message{
+				Record: map[string]any {
+					"tag":     insightsmetricstag,
+					"messages": insightsmetricsmessages,
+				},
+				Time: emitTime,
+			}
+
+			FLBLogger.Print("perf::emitted ", len(insightsmetricsmessages) ," insights metrics records @ ", time.Now().UTC().Format(time.RFC3339))
+
+			timeDifference := int(math.Abs(float64(time.Now().Unix() - telemetryTimeTracker)))
```

Comment:
i see telemetryTimeTracker being reset every at line 93 in this code. Please double check on this.